src/metabase/lib/js.cljs

```diff
@@ -556,6 +556,30 @@
   [a-query]
   (lib.core/TemplateTags-> (lib.core/template-tags a-query)))
 
+(defn ^:export does-require-native-collection
+  "Returns whether this database requires a native colletion."
+  [database-id metadata]
+  (lib.core/requires-native-collection? (metadataProvider database-id metadata)))
+
+(defn ^:export with-native-collection
+  "Changes the db collection to run this query against.
+   The first stage must be a native type. The database must support `native-specified-collection`"
```

Comment:
Cannot we ignore the collection if the DB doesn't support/require it? That might simplify the caller's life.